[bug]  Type Mismatch in Spec Assembly Causes Build Failure in SpecAppService

# 1. Describe the bug
The backend fails to compile due to a type mismatch in the spec assembly logic. In `SpecAppService.cs`, a dictionary that maps section item positions to a single `AssembledSectionItemDto` is being assigned where a dictionary mapping positions to a **list** of `AssembledSectionItemDto` is expected.

This results in a compile-time error and blocks further development related to spec assembly, dependency graph traversal, and topological ordering.

## Source:
`/src/SeeSpec.Application/Services/SpecService/SpecAppService.cs`

Error location:
- Line 432

Compiler error: 
CS0029: Cannot implicitly convert type
System.Collections.Generic.Dictionary<int, AssembledSectionItemDto>
to
System.Collections.Generic.Dictionary<int, List> 

# 2. To Reproduce

## Set up process if present:
- .NET backend project (`SeeSpec.Application`)
- Standard build configuration

## Steps to reproduce the behavior:
1. Run `dotnet build`
2. Build fails in `SpecAppService.cs`
3. Compiler error CS0029 is raised at line 432

# 3. Expected behavior:
- Section item assembly uses a consistent and correct dictionary type
- The backend compiles successfully
- Deterministic section item grouping and ordering remain intact

# 4. Screenshots
N/A

# 5. Machine specifications:
- Reproducible across all development machines
- OS-independent

# 6. Additional context
This issue was introduced while extending the existing spec assembly logic to support runtime section dependency graphs and topological ordering. The in-memory grouping logic was updated, but the dictionary value type was not aligned with the expected DTO structure.

--- a/aspnet-core/src/SeeSpec.Application/Services/SpecService/SpecAppService.cs
+++ b/aspnet-core/src/SeeSpec.Application/Services/SpecService/SpecAppService.cs
@@ -429,7 +429,7 @@ namespace SeeSpec.Services.SpecService
                     Section = section,
                     ItemsByPosition = itemsBySectionId.TryGetValue(section.Id, out var groupedItems)
                         ? BuildItemDictionary(groupedItems)
-                        : new Dictionary<int, AssembledSectionItemDto>()
+                        : new Dictionary<int, List<AssembledSectionItemDto>>()
                 });
 
             foreach (var dependency in sectionDependencies)

